pkg/endpoint/endpoint.go

```diff
@@ -163,6 +177,36 @@ func init() {
 	}
 }
 
+// getCiliumClient builds and returns a k8s auto-generated client for cilium
+// objects
+func getCiliumClient() (ciliumClient cilium_client_v2.CiliumV2Interface, err error) {
+	// This allows us to reuse the k8s client
+	ciliumEndpointSyncControllerOnce.Do(func() {
+		var (
+			restConfig *rest.Config
+			k8sClient  *clientset.Clientset
+		)
+
+		restConfig, err = k8s.CreateConfig()
+		if err != nil {
+			return
+		}
+
+		k8sClient, err = clientset.NewForConfig(restConfig)
+		if err != nil {
+			return
+		}
+
+		ciliumEndpointSyncControllerK8sClient = k8sClient
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return ciliumEndpointSyncControllerK8sClient.CiliumV2(), nil
```

Comment:
Is this intended? What happens if we are running k8s 1.6 and the cilium client should be `CiliumV1()`?

Reply:
We wouldn't have the CRD in the 1.6 case (I didn't create a TPR). I'll check the version and return an error, and that should avoid the impossible situation.

Reply:
That's true